File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while len(scoville)>=2 and scoville[0] < K:
        first = heapq.heappop(scoville)
        second = heapq.heappop(scoville)
        heapq.heappush(scoville, first+second*2)
        answer+=1
        # print(scoville)
    if scoville[0]<K: 
        return -1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> pq = new PriorityQueue<>((o1, o2)-> {
            return o1 - o2;
        });
        for (int s: scoville){
            pq.offer(s);
        }
        
        while (pq.size() > 1){
            // System.out.println(pq);
            if (pq.peek() < K){
                int first = pq.poll();
                int second = pq.poll();
                pq.offer(first + second*2);
                answer += 1;
            } else {
                break;
            }
        }
        if (pq.peek() < K){
            return -1;
        } else {
            return answer;
        }
    }
}